lib/cib/cib_attrs.c

```diff
@@ -86,10 +93,12 @@ find_nvpair_attr_delegate(cib_t * the_cib, const char *attr, const char *section
     }
 
     xpath_string = calloc(1, xpath_max);
-    offset += snprintf(xpath_string + offset, xpath_max - offset, "%.128s", get_object_path(section));
```

Comment:
Are we discussing aesthetics or a sound code?

```
$ CIB_file=cib-0733c987.xml valgrind crm_attribute --query -i "$(python -c 'print "X" * 960')"
[...]
==2701== Invalid write of size 1
==2701==    at 0x30BEA771B2: vsnprintf (vsnprintf.c:117)
==2701==    by 0x30BEA540D1: snprintf (snprintf.c:33)
==2701==    by 0x3016C10446: find_nvpair_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x3016C115D6: read_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x4015CA: ??? (in /usr/sbin/crm_attribute)
==2701==    by 0x30BEA2078F: (below main) (libc-start.c:289)
==2701==  Address 0x5398e43 is 3 bytes after a block of size 1,024 alloc'd
==2701==    at 0x4A08987: calloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==2701==    by 0x3016C102D5: find_nvpair_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x3016C115D6: read_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x4015CA: ??? (in /usr/sbin/crm_attribute)
==2701==    by 0x30BEA2078F: (below main) (libc-start.c:289)
==2701== 
==2701== Invalid write of size 1
==2701==    at 0x30BEA7B92B: _IO_default_xsputn (genops.c:475)
==2701==    by 0x30BEA496CC: vfprintf (vfprintf.c:1323)
==2701==    by 0x30BEA771C8: vsnprintf (vsnprintf.c:119)
==2701==    by 0x30BEA540D1: snprintf (snprintf.c:33)
==2701==    by 0x3016C10446: find_nvpair_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x3016C115D6: read_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x4015CA: ??? (in /usr/sbin/crm_attribute)
==2701==    by 0x30BEA2078F: (below main) (libc-start.c:289)
==2701==  Address 0x5398e43 is 3 bytes after a block of size 1,024 alloc'd
==2701==    at 0x4A08987: calloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==2701==    by 0x3016C102D5: find_nvpair_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x3016C115D6: read_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x4015CA: ??? (in /usr/sbin/crm_attribute)
==2701==    by 0x30BEA2078F: (below main) (libc-start.c:289)
==2701== 
==2701== Invalid write of size 1
==2701==    at 0x30BEA771DC: vsnprintf (vsnprintf.c:122)
==2701==    by 0x30BEA540D1: snprintf (snprintf.c:33)
==2701==    by 0x3016C10446: find_nvpair_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x3016C115D6: read_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x4015CA: ??? (in /usr/sbin/crm_attribute)
==2701==    by 0x30BEA2078F: (below main) (libc-start.c:289)
==2701==  Address 0x5398e44 is 4 bytes after a block of size 1,024 alloc'd
==2701==    at 0x4A08987: calloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==2701==    by 0x3016C102D5: find_nvpair_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x3016C115D6: read_attr_delegate (in /usr/lib64/libcib.so.4.0.0)
==2701==    by 0x4015CA: ??? (in /usr/sbin/crm_attribute)
==2701==    by 0x30BEA2078F: (below main) (libc-start.c:289)
[...]
```
Macro definition can be changed to resemble the original code more, though.